SalesforceCommerceCloud/pwa-kit
# [App Extensibility] Allow app extensions to provide their default config (@W-17336560@)

This PR allows app extensions to provide their default config, and the `extension-sdk` will merge this default and the user-defined configs for them.

The user-defined config, in this case, is the one coming from the base project. It'll be merged with each extension's default config. This merge happens right before the call to `renderTemplate`, so the handlebar template would already have the complete configs.

The other main changes are for updating all of our extensions to have default config. The default config is stored in `config/default.json`. It may have some placeholder values (e.g. <CLIENT_ID>) if the user has to be the one providing them. That's why there is also the `config/sample.json` to show a working valid config that others can use.

- [x] Do a _deep_ merge of the configs
- [x] Write, update the tests

## How to test-drive the PR

Run the typescript minimal project that uses the sample, store-locator, and chakra-storefront extensions.

0. Revert the commit ee13cc4cdcbb18f9a246f8141e56185436f5e33a
1. `npm ci` at the monorepo root
2. `npm start` at the typescript-minimal directory
3. Verify in your browser that the project loads with the storefront (and red border around the app)
4. Navigate to `/store-locator` and verify that the store-locator loads as well.
